build.gradle doesn't support AGP 8+

**Describe the bug**
After running the AGP upgrader in Android Studio and trying to start a debug session I get:
```
Launching lib/main_development.dart on LEMP in debug mode...
Running Gradle task 'assembleDevelopmentDebug'...

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':heart_rate_flutter'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file: /home/csaba/.pub-cache/hosted/pub.dev/heart_rate_flutter-0.0.2/android/build.gradle. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 12s
Error: Gradle task assembleDevelopmentDebug failed with exit code 1
```

The reason for this is that AGP 8+ requires a namespace attribute in the build.
Flutter Community projects for example solve this by adding a conditional clause to the build.gradle, this way it also remain pre AGP 8 compatible:

```groovy
android {
    // Conditional for compatibility with older AGP.
    if (project.android.hasProperty("namespace")) {
        namespace 'com.example.heart_rate_flutter'
    }
```

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -22,6 +22,11 @@ rootProject.allprojects {
 apply plugin: 'com.android.library'
 
 android {
+    // Conditional for compatibility with older AGP.
+    if (project.android.hasProperty("namespace")) {
+        namespace 'com.example.heart_rate_flutter'
+    }
+
     compileSdkVersion 31
 
     compileOptions {

